analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +29,80 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
+    {
+        return IsAccessibleFromOtherAssemblies(Node)
+               && (IsSupportedMethod(Node) || IsSupportedPropertyAccessor(Node));
+
+        static bool IsSupportedMethod(SyntaxNode node) =>
+            node is BaseMethodDeclarationSyntax { ParameterList.Parameters.Count: > 0 } method
+            && MethodDereferencesArguments(method);
+
+        static bool IsSupportedPropertyAccessor(SyntaxNode node) =>
+            node is AccessorDeclarationSyntax { RawKind: (int)SyntaxKind.SetAccessorDeclaration or (int)SyntaxKindEx.InitAccessorDeclaration } accessor
```

Comment:
There is an extension method of `SyntaxNode` which does that:
```cs
public static bool IsAnyKind(this SyntaxNode syntaxNode, params SyntaxKind[] syntaxKinds) =>
            syntaxNode != null && syntaxKinds.Contains((SyntaxKind)syntaxNode.RawKind);
```

I would probably use that, instead of deconstructing on `RawKind`, which seems a bit like looking into the "innards" of `SyntaxNode`.

Reply:
Noted. I removed the pattern matching, so it's no longer needed.